dev-tools/packaging/package_test.go

```diff
@@ -740,22 +747,17 @@ func readDocker(dockerFile string) (*packageFile, *dockerInfo, error) {
 		}
 
 		switch {
-		case header.Name == "manifest.json":
-			manifest, err = readDockerManifest(tarReader)
-			if err != nil {
-				return nil, nil, err
-			}
-		case strings.HasSuffix(header.Name, ".json") && header.Name != "manifest.json":
+		case header.Name == manifest.Config:
 			info, err = readDockerInfo(tarReader)
 			if err != nil {
 				return nil, nil, err
 			}
-		case strings.HasSuffix(header.Name, "/layer.tar"):
+		case slices.Contains(manifest.Layers, header.Name):
```

Comment:
This isn't in 8.12 yet since it is still at Go 1.20, so is 7.17 but we plan to change that with https://github.com/elastic/beats/pull/37694 since 7.17.18 will be after the Go 1.22 release.

I don't think we can put it into 8.12 since Go 1.21 modifies our support matrix by dropping Windows 8.1

Reply:
When we do the backport for 8.12 we can substitute in `golang.org/x/exp/slices` or a helper function or a map.